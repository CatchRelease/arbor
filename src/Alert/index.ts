export { default as BannerAlert } from './BannerAlert';
export { default as CardAlert } from './CardAlert';
export { default as InlineAlert } from './InlineAlert';
export { default as PaneAlert } from './PaneAlert';
export { default as ToastAlert } from './ToastAlert';
