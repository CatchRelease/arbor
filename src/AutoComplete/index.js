export { default } from './AutoComplete';
export { default as styledAutoCompleteComponents } from './styledAutoCompleteComponents';
export { default as StyledAutoComplete } from './StyledAutoComplete';
