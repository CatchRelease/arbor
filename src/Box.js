import styled from '@emotion/styled';
import {
  boxShadow,
  color,
  display,
  space,
  width,
  height,
  maxWidth,
  textAlign
} from 'styled-system';

const Box = styled.div`
  box-sizing: border-box;
  ${boxShadow};
  ${color};
  ${display};
  ${height};
  ${maxWidth};
  ${space};
  ${textAlign};
  ${width};
`;

Box.propTypes = {
  ...boxShadow.propTypes,
  ...color.propTypes,
  ...display.propTypes,
  ...height.propTypes,
  ...maxWidth.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
  ...width.propTypes
};

Box.section = Box.withComponent('section');

export default Box;
