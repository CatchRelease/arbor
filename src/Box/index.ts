export { default, Props as BoxProps } from './Box';
