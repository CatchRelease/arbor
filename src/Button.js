import styled from 'react-emotion';
import PropTypes from 'prop-types';
import { transparentize } from 'polished';
import {
  borderRadius,
  brandFont,
  colors,
  fontWeights,
  lineHeights,
  spacings,
  typography
} from './theme';

const Button = styled.button`
  font-family: ${brandFont};
  font-size: ${typography.text.ui.fontSize};
  border: none;
  color: ${colors.white};
  font-weight: ${fontWeights.medium};
  line-height: ${lineHeights.large};
  outline: none;
  transition: box-shadow 0.3s ease, background 0.2s ease;

  &:enabled {
    cursor: pointer;
  }
`;

Button.CTA = styled(Button.withComponent('button'))`
  background: ${colors.blue};
  border-radius: ${borderRadius.large};
  box-shadow: 0 1px 3px ${transparentize(0.7, colors.blue)},
    inset 0 -1px 0 ${transparentize(0.9, colors.black)};
  padding: ${spacings.small} 0;
  text-transform: uppercase;
  width: 100%;

  &:disabled {
    background: ${colors.grey80};
    box-shadow: none;
  }

  &:hover:enabled {
    background: ${colors.blueLight};
    box-shadow: 0 2px 6px ${transparentize(0.8, colors.blueLight)};
  }

  &:focus:enabled {
    background: ${colors.blue};
    box-shadow: 0 0 0 1px ${colors.white}, ${colors.blueDark} 0 0 0 2px;
  }
`;

Button.CTA.displayName = 'Button.CTA';

Button.CTA.propTypes = {
  disabled: PropTypes.bool
};

Button.CTA.defaultProps = {
  disabled: false
};

export default Button;
