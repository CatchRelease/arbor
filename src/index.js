export * from './theme';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Card } from './Card';
export { default as CardActions } from './CardActions';
export { default as CardPreview } from './CardPreview';
export { default as CardRow } from './CardRow';
export { default as Checkbox } from './Checkbox';
export { default as Dropdown } from './Dropdown';
export { default as Flex } from './Flex';
export { default as Grid } from './Grid';
export { default as Heading } from './Heading';
export { default as Icon } from './Icon';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Link } from './Link';
export { default as Menu } from './Menu';
export { default as Pagination } from './Pagination';
export { default as Paragraph } from './Paragraph';
export { default as Popover } from './Popover';
export { default as Text } from './Text';
export { default as Tooltip } from './Tooltip';

export { default as reset } from './reset';
export { default as theme } from './theme';
