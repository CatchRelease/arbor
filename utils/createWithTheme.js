import React from 'react';
import { CacheProvider } from '@emotion/core';
import createCache from '@emotion/cache';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';
import theme from '../src/theme';

const myCache = createCache();
myCache.compat = true;

export default (Component) =>
  create(
    <CacheProvider value={myCache}>
      <ThemeProvider theme={theme}>{Component}</ThemeProvider>
    </CacheProvider>
  );
