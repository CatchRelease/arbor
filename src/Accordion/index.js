import styled from '@emotion/styled';

import Box from '../Box';

const Accordion = styled(Box)`
  border: 1px solid ${props => props.theme.colors.border.default};
  border-radius: ${props => props.theme.borderRadius.large};
`;

export default Accordion;
