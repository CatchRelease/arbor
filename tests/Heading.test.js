import React from 'react';
import createWithTheme from './utils/createWithTheme';

import { Heading } from '../src';

['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].forEach(variant => {
  test(`renders a ${variant}`, () => {
    const Variant = Heading[variant];
    const tree = createWithTheme(<Variant>Hello World</Variant>);

    expect(tree).toMatchSnapshot();
  });
});
