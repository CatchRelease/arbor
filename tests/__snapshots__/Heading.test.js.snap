// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a h1 1`] = `
.emotion-0 {
  font-family: 'Open Sans';
  font-weight: 700;
  line-height: 1.25;
  color: #222222;
  font-size: 2.125rem;
  margin-bottom: calc(2.125rem * 0.5);
}

<h1
  className="emotion-0 emotion-1"
  color="grey100"
>
  Hello World
</h1>
`;

exports[`renders a h2 1`] = `
.emotion-0 {
  font-family: 'Open Sans';
  font-weight: 700;
  line-height: 1.25;
  color: #666666;
  font-size: 1.75rem;
  margin-bottom: calc(1.75rem * 0.5);
}

<h2
  className="emotion-0 emotion-1"
  color="grey80"
>
  Hello World
</h2>
`;

exports[`renders a h3 1`] = `
.emotion-0 {
  font-family: 'Open Sans';
  font-weight: 700;
  line-height: 1.25;
  color: #666666;
  font-size: 1.375rem;
  margin-bottom: calc(1.375rem * 0.5);
}

<h3
  className="emotion-0 emotion-1"
  color="grey80"
>
  Hello World
</h3>
`;

exports[`renders a h4 1`] = `
.emotion-0 {
  font-family: 'Open Sans';
  font-weight: 700;
  line-height: 1.25;
  color: #666666;
  font-size: 1.125rem;
  margin-bottom: calc(1.125rem * 0.5);
}

<h4
  className="emotion-0 emotion-1"
  color="grey80"
>
  Hello World
</h4>
`;

exports[`renders a h5 1`] = `
.emotion-0 {
  font-family: 'Open Sans';
  font-weight: 700;
  line-height: 1.25;
  color: #222222;
  font-size: 0.875rem;
  margin-bottom: calc(0.875rem * 0.5);
  font-weight: 600;
  line-height: 1.5;
}

<h5
  className="emotion-0 emotion-1"
  color="grey100"
>
  Hello World
</h5>
`;

exports[`renders a h6 1`] = `
.emotion-0 {
  font-family: 'Open Sans';
  font-weight: 700;
  line-height: 1.25;
  color: #666666;
  font-size: 0.875rem;
  margin-bottom: calc(0.875rem * 0.5);
  font-weight: 400;
  line-height: 1.5;
}

<h6
  className="emotion-0 emotion-1"
  color="grey80"
>
  Hello World
</h6>
`;
